webapp/src/main/java/org/apache/atlas/web/filters/AtlasXSSPreventionFilter.java

```diff
@@ -0,0 +1,208 @@
+                .allowAttributes("valign").matching(REGEX_CELL_VERTICAL_ALIGN).onElements("col", "colgroup", "td", "th", "tbody", "tfoot")
+                .allowAttributes("abbr").matching(REGEX_PARAGRAPH).onElements("td", "th")
+                .allowAttributes("colspan","rowspan").matching(REGEX_INTEGER).onElements("td", "th")
+                .allowAttributes("headers").matching(REGEX_SPACE_SEPARATED_TOKENS).onElements("td", "th")
+                .allowAttributes("scope").matching(REGEX_SCOPE).onElements("td", "th")
+                .allowAttributes("nowrap").matching(REGEX_NOWRAP).onElements("td", "th")
+                //allow Forms
+                //By and large, forms are not permitted. However there are some form
+                // elements that can be used to present data, and we do permit those
+                .allowElements("meter", "progress")
+                .allowAttributes("value", "min", "max", "low", "high", "optimum").matching(REGEX_NUMBER).onElements("meter", "progress")
+                .allowAttributes("value","max").matching(REGEX_NUMBER).onElements("progress")
+                //Allow Image
+                .allowElements("img")
+                .allowAttributes("align").matching(REGEX_IMAGE_ALIGNMENT).onElements("img")
+                .allowAttributes("alt").matching(REGEX_PARAGRAPH).onElements("img")
+                .allowAttributes("height", "width").matching(REGEX_NUMBER_OR_PERCENT).onElements("img")
+                .allowAttributes("src").matching(REGEX_ON_OFFSITE_URL).onElements("img")
+                //Allow Global Attributes
+                .allowAttributes("dir").matching(REGEX_DIRECTION).globally()
+                .allowAttributes("lang").matching(REGEX_LANG).globally()
+                .allowAttributes("id").matching(REGEX_ID).globally()
+                .allowAttributes("title").matching(REGEX_PARAGRAPH).globally()
+
+                .allowWithoutAttributes("a", "span")
+                .toFactory()
+                .and(Sanitizers.IMAGES);
+
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+        HttpServletResponse response = (HttpServletResponse) servletResponse;
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+
+        String serverName = request.getServerName();
+        if (StringUtils.isNotEmpty(serverName) && serverName.contains(AtlasConfiguration.REST_API_XSS_FILTER_EXLUDE_SERVER_NAME.getString())) {
+            LOG.debug("AtlasXSSPreventionFilter: skipping filter for serverName: {}", serverName);
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        String method = request.getMethod();
+        if(!method.equals("POST") && !method.equals("PUT")) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        String contentType = request.getContentType();
```

Comment:
Why do we need this check?

Reply:
With POST and PUT we don't really use other format like text, HTML and XML(Checked in all available API's in Atlas). So if user send those it will automatically get rejected no need to waste processing power on processing those unwanted content.